src/libostree/ostree-repo.c

```diff
@@ -2691,7 +2691,14 @@ min_free_space_size_validate_and_convert (OstreeRepo    *self,
         g_assert_not_reached ();
     }
 
-  self->min_free_space_mb = g_ascii_strtoull (size_str, NULL, 10) << shifts;
+  guint64 min_free_space = g_ascii_strtoull (size_str, NULL, 10);
+  if (shifts > 0 && g_bit_nth_lsf (min_free_space, 64 - shifts) != -1)
```

Comment:
Neat! IIUC, shouldn't this be `63 - shifts`? E.g. if `shifts == 10`, then the index of the largest bit allowed to be 1 is 53 (so that after the shift it's at 63). And from the documentation on `g_bit_nth_lsf`, it doesn't include the index we pass, so if we want it to start searching from 54, we should pass 53, right?

Reply:
You are so right. Fixes coming right up! Thank you for noticing. Apparently I calculated `0` as the MSB and upwards
 :-(